### PR TITLE
Fix saved search alert notification URL

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #12212

This should fix an issue with the encoding of the inner query parameters.

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -126,7 +126,8 @@
 
     // redirect to ticket
     if (isset($_GET["redirect"])) {
-        Toolbox::manageRedirect($_GET["redirect"]);
+        global $_UGET;
+        Toolbox::manageRedirect($_UGET["redirect"]);
```
Comment (cedric-anne) [inline index.php:130]:
```suggestion
        Toolbox::manageRedirect(Sanitizer::unsanitize($_UGET["redirect"]));
```

Comment (cconard96) [reply index.php:130]: Is this change meant to unsanitize the GET property instead of the UGET one?

Comment (cedric-anne) [reply index.php:130]: Indeed, you should unsanitize `$_GET`.
`$_U*` was introduced before sanitizer could perform a reverse non-destructive sanitization. These should not be used anymore.

**Review by cedric-anne — COMMENTED** [review]
I added commit to try to handle more issues and prevent having to use double urlencode.

**Comment by cconard96** [comment]
I re-tested with the latest commits and the original issue still seems fixed.

**Comment by cconard96** [comment]
The original issue is back with the last 2 commits.
Example resulting URL:
`/index.php?redirect=/front/savedsearch.php?action=load&id=40&noAUTO=1`
The redirect then goes to:
`/front/savedsearch.php?action=load`

The query parameters meant for the target page `/front/savedsearch.php` have to be encoded until the redirect takes place.

**Comment by cedric-anne** [comment]
My bad. I tested my fix without using the `formatUrl` method.

I added a commit to remove the call to `urldecode` in this method. `$redirect` parameter value have to be url-encoded, so decoding the whole chain requires to have a it double encoded in method caller, which make no sense to me.

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Comment by cedric-anne** [comment]
@cconard96 Is it OK for you now ?

**Comment by cconard96** [comment]
Yes, this seems to be working now.

**Pull request merged**